drgn/helpers/linux/sched.py

```diff
@@ -163,3 +164,44 @@ def loadavg(prog: Program) -> Tuple[float, float, float]:
     avenrun = prog["avenrun"]
     vals = [avenrun[i].value_() / (1 << 11) for i in range(3)]
     return (vals[0], vals[1], vals[2])
+
+
+def cpu_rq(prog: Program, cpu: int) -> Object:
+    """
+    Get the runqueue for a given cpu.
+
+    :param cpu: CPU number.
+    :returns: ``struct rq``
+    """
+    return per_cpu(prog["runqueues"], cpu)
+
+
+def task_rq(task: Object) -> Object:
+    """
+    Get the runqueue for a given task.
+
+    :param task: ``struct task_struct *``
+    :returns: ``struct rq``
+    """
+    return cpu_rq(task.prog_, task_cpu(task))
+
+
+def task_lastrun2now(task: Object) -> int:
+    """
+    Get the number of nanoseconds since a task last started running.
+
+    The return duration will cover task's last run time on cpu and also
+    the time staying in current status, usually the time slice for task
+    on cpu will be short, so this can roughly tell how long this task
+    has been staying in current status.
+    For task status in "R" status, if it's still on cpu, then this return
+    the duration time this task has been running, otherwise it roughly tell
+    how long this task has been staying in runqueue.
+
+    :param task: ``struct task_struct *``
+    :returns: duration in ns granularity
+    """
+    arrival_time = task.sched_info.last_arrival.value_()
```

Comment:
Running the tests revealed that this only exists if `CONFIG_SCHED_INFO=y` (selected by `CONFIG_SCHEDSTATS` or `CONFIG_TASK_DELAY_ACCT`). I'm rebuilding our test kernels now to enable this option: https://github.com/osandov/drgn/commit/05ed1c947969c73bf3610ed37190775fa0ed683e, https://github.com/osandov/drgn/actions/runs/16479745871. But please mention this requirement in the docstring.